### ISSUE TITLE
XSS Java not detected in a simple example

### ISSUE THREAD

**Issue by giper45** [github/codeql#10395]
Hello, I would like to have a support for the following example. 
I have created the following Servlet PoC vulnerable to cross-site scripting: 
* https://github.com/giper45/simple-servlet.git 

that contains the Xss.java example  in the CWE-079 folder of codeql: 
* https://github.com/github/codeql/blob/main/java/ql/src/Security/CWE/CWE-079/XSS.java

The vulnerable code is located in the Xss.java folder:   
https://github.com/giper45/simple-servlet/blob/master/src/main/java/Xss.java

I clone the project:   
```
git clone https://github.com/giper45/simple-servlet
cd simple-servlet
```

I create the database: 
``` 
codeql database create --language=java java-database.db
```
Then, I download the codeql libraries:  
```  
cd ~/bin
git clone https://github.com/github/codeql.git codeql-libraries
``` 

Finally, I run the query to detect the cross-site scripting vulnerability: 
``` 
codeql query run ~/bin/codeql-libraries/java/ql/src/Security/CWE/CWE-079/XSS.ql --database java-database.db
```

Why the query does not detect the vulnerability?   
Thank you in advance.

**Comment by atorralba** [comment]
Hi @giper45! It seems that you found an outdated example in our XSS query.

`HttpServletResponse.sendError` is not currently being considered a XSS sink in CodeQL after https://github.com/github/codeql/pull/5940, since Servlet API implementations generally already escape the error message.

To find valid examples of alerts raised by our XSS query, you can refer to the [tests](https://github.com/github/codeql/blob/097d5189e9bbaecfd0e239e5198ac8e8ac14f2f6/java/ql/test/query-tests/security/CWE-079/semmle/tests/XSS.java).

Thanks for pointing this out! We'll fix the example so that it shows a current example of the vulnerability.

**Comment by giper45** [comment]
In fact, the example was a false positive in modern Tomcat environments, but I was thinking that it should have triggered it anyway for the static analysis. Thank you very much for the clarification!

**Comment by atorralba** [comment]
Fixed in #10408, closing this. Thanks @giper45.
